internal/argotunnel/translator.go

```diff
@@ -274,6 +278,37 @@ func (t *syncTranslator) getVerifiedCert(namespace, name, host string) (cert []b
 		return
 	}
 
+	return
+}
+
+func (t *syncTranslator) findVerifiedCertSecret(candidates []resource, host string) (resource, error) {
+	for _, secret := range candidates {
+
+		cert, exists, _ := t.getCert(secret.namespace, secret.name)
+		if !exists {
+			continue
+		}
+
+		err := verifyCertForHost(cert, host)
```

Comment:
A user should be able to identify which secret he/she needs for the ingress, rather than relying on an expensive process to attempt to match a cert to a hostname.  This moves from a single verification to avoid network and tunnel creation failure to N verifications (degrading processing for any tunnel that uses default secrets, and further degrading as more defaults secrets are used).